src/tests/ftest/util/ior_test_base.py

```diff
@@ -285,97 +288,115 @@ def stop_ior(self):
-        """Run the IOR command.
+    def run_custom_ior_cmd(self, ior_command, clients, results, job_num,
+                           intercept=None):
+        """Run customized IOR command, not self.ior_cmd.
+
+        Expected to be used with a threaded code where multiple IOR commands are
+        executed in parallel.
+
+        Display pool space before running it for a reference.
 
         Args:
+            ior_command (IorCommand): Custom IOR command instance.
             clients (list): hosts on which to run ior
             results (dict): A dictionary object to store the ior metrics
             job_num (int): Assigned job number
             intercept (str, optional): path to interception library. Defaults to
                 None.
         """
-        self.lock.acquire(True)
-        tsize = self.ior_cmd.transfer_size.value
-        testfile = os.path.join(self.dfuse.mount_dir.value,
-                                "testfile{}{}".format(tsize, job_num))
+        self.log.info("--- IOR Thread {}: Start ---".format(job_num))
+        tsize = ior_command.transfer_size.value
+        testfile = os.path.join(
+            self.dfuse.mount_dir.value, "testfile{}{}".format(tsize, job_num))
         if intercept:
             testfile += "intercept"
-        self.ior_cmd.test_file.update(testfile)
-        manager = self.get_ior_job_manager_command()
+        ior_command.test_file.update(testfile)
+
+        # Get the custom job manager that's associated with this thread.
+        manager = self.get_ior_job_manager_command(custom_ior_cmd=ior_command)
+
         procs = (self.processes // len(self.hostlist_clients)) * len(clients)
-        env = self.ior_cmd.get_default_env(str(manager), self.client_log)
+        env = ior_command.get_default_env(str(manager), self.client_log)
         if intercept:
             env["LD_PRELOAD"] = intercept
         manager.assign_hosts(
-            clients,
-            self.workdir,
-            self.hostfile_clients_slots)
+            clients, self.workdir, self.hostfile_clients_slots)
         manager.assign_processes(procs)
         manager.assign_environment(env)
-        self.lock.release()
+        self.display_pool_space()
+
+        self.log.info("--- IOR Thread {}: Starting IOR ---".format(job_num))
         try:
-            self.display_pool_space()
-            out = manager.run()
-            self.lock.acquire(True)
-            results[job_num] = IorCommand.get_ior_metrics(out)
-            self.lock.release()
+            ior_output = manager.run()
+            results[job_num] = IorCommand.get_ior_metrics(ior_output)
         except CommandFailure as error:
             self.log.error("IOR Failed: %s", str(error))
-            self.fail("Test was expected to pass but it failed.\n")
+            self.fail("IOR thread failed!")
         finally:
             self.display_pool_space()
 
+        self.log.info("--- IOR Thread {}: End ---".format(job_num))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions